sys/riscv/riscv/pmap.c

```diff
@@ -3774,6 +3782,11 @@ pmap_caploadgen_update(pmap_t pmap, vm_offset_t va, vm_page_t *mp, int flags)
 						sfence_vma_page(va);
 					}
 					PMAP_UNLOCK(pmap);
+					/*
+					 * XXXMJ don't we need to hold a
+					 * reference to m?  What prevents
+					 * concurrent reclamation?
+					 */
```

Comment:
I _believe_, or at some point in the past believed, I think, that if we got here the page has to have been wired by the caller through the `PMAP_CAPLOADGEN_WIRE` behaviour.  Even if that isn't true in the current code, we probably should make this API behave as though that flag were always passed to work around the `m->object != obj` bug you identified below, so that should mean that `m` is always `vm_page_wire`-d here, yes?

Reply:
I think we could correct assert `vm_page_wired(m)` here if all callers of `pmap_caploadgen_update()` specify `PMAP_CAPLOADGEN_WIRE`. Indeed, I think that `pmap_caploadgen_update()` should unconditionally wire the page.